jerry-core/ecma/builtin-objects/ecma-builtin-helpers-date.c

```diff
@@ -153,13 +92,10 @@ ecma_date_day_from_year (ecma_number_t year) /**< year value */
  *
  * @return  time value of the start of a year
  */
-inline ecma_number_t __attr_always_inline___
+static inline ecma_number_t __attr_always_inline___
 ecma_date_time_from_year (ecma_number_t year) /**< year value */
 {
-  if (ecma_number_is_nan (year))
-  {
-    return year; /* year is NaN */
-  }
+  JERRY_ASSERT (!ecma_number_is_nan (year));
```

Comment:
This means NaN is not a valid input. Are you sure about this? I thought we should return NaN if the input is NaN. Please check test262 on a debug build of JerryScript.